node/src/signature_mngr.rs

```diff
@@ -579,91 +580,19 @@ impl Actor for SignatureManagerAdapter {
-
-    fn handle(&mut self, msg: SetBn256Key, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<Sign> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<KeyedSignature, failure::Error>>;
-
-    fn handle(&mut self, msg: Sign, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<Bn256Sign> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<Bn256KeyedSignature, failure::Error>>;
-
-    fn handle(&mut self, msg: Bn256Sign, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<GetPkh> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<PublicKeyHash, failure::Error>>;
-
-    fn handle(&mut self, msg: GetPkh, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<GetPublicKey> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<PublicKey, failure::Error>>;
-
-    fn handle(&mut self, msg: GetPublicKey, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<GetBn256PublicKey> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<Bn256PublicKey, failure::Error>>;
-
-    fn handle(&mut self, msg: GetBn256PublicKey, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<GetKeyPair> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<(ExtendedPK, ExtendedSK), failure::Error>>;
-
-    fn handle(&mut self, msg: GetKeyPair, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<GetBn256KeyPair> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<(Bn256PublicKey, Bn256SecretKey), failure::Error>>;
-
-    fn handle(&mut self, msg: GetBn256KeyPair, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<VrfProve> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<(VrfProof, Hash), failure::Error>>;
-
-    fn handle(&mut self, msg: VrfProve, _ctx: &mut Self::Context) -> Self::Result {
-        Box::pin(self.crypto.send(msg).flatten_err())
-    }
-}
-
-impl Handler<VerifySignatures> for SignatureManagerAdapter {
-    type Result = ResponseFuture<Result<(), failure::Error>>;
+// Delegate all the SignatureManager messages to the inner SignatureManager
+impl<M> Handler<M> for SignatureManagerAdapter
+where
+    M: Message + Send + 'static,
+    <M as actix::Message>::Result: Send,
+    Result<<M as actix::Message>::Result, actix::MailboxError>:
+        FlattenResult<OutputResult = <M as actix::Message>::Result>,
+    SignatureManager: Handler<M>,
```

Comment:
If possible I would like this to be simplified as:

```suggestion
    DelegateHandler<M, SignatureManager>
```

Reply:
Not possible :disappointed: